pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +364,353 @@ def Sphere(
+        points. If specified, it overrides any choices for ``outer_radius``,
+        ``inner_radius``, and ``radius_resolution``.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying theta
+        points in degrees. If specified, it overrides any choices for
+        ``start_theta``, ``end_theta``, and ``theta_resolution``.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying phi
+        points in degrees. If specified, it overrides any choices for
+        ``start_phi``, ``end_phi``, and ``phi_resolution``.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees latitude.
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Solid sphere mesh.
+
+    See Also
+    --------
+    pyvista.Sphere
+
+    Examples
+    --------
+    Create a solid sphere.
+
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> solid_sphere = pv.SolidSphere()
+    >>> solid_sphere.plot(show_edges=True)
+
+    A solid sphere is 3D in comparison to the 2d `pyvista.Sphere`.
+    Generate a solid hemi-sphere to see the internal structure.
+
+    >>> isinstance(solid_sphere, pv.UnstructuredGrid)
+    True
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180, end_theta=360
+    ... )
+    >>> partial_solid_sphere.plot(show_edges=True)
+
+    To see the cell structure inside the solid sphere,
+    only 1/4 of the sphere is generated. The cells exploded
+    and colored by radial position.
+
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180,
+    ...     end_theta=360,
+    ...     start_phi=0,
+    ...     end_phi=90,
+    ...     radius_resolution=5,
+    ...     theta_resolution=8,
+    ...     phi_resolution=8,
+    ... )
+    >>> partial_solid_sphere["cell_radial_pos"] = np.linalg.norm(
+    ...     partial_solid_sphere.cell_centers().points, axis=-1
+    ... )
+    >>> partial_solid_sphere.explode(1).plot()
+
+    """
+    if radius is None:
+        radius = np.linspace(inner_radius, outer_radius, radius_resolution)
+    if theta is None:
+        theta = np.linspace(start_theta, end_theta, theta_resolution)
+    if phi is None:
+        phi = np.linspace(start_phi, end_phi, phi_resolution)
+
+    if radius[0] < 0.0:
+        raise ValueError("minimum radius cannot be negative")
+    if theta[0] < 0.0:
+        raise ValueError("minimum theta cannot be negative")
+    if phi[0] < 0.0:
+        raise ValueError("minimum phi cannot be negative")
+    if theta[-1] > 360.0:
+        raise ValueError("maximum theta cannot be > 360.0")
+    if phi[-1] > 180.0:
+        raise ValueError("maximum phi cannot be > 180.0")
```

Comment:
1. It's a bit of a pain in the ass, but I think we should document these constraints with the respective parameters.
2. For user-supplied arrays we can only hope that the first/last values are min/max. It might be worth sorting the values first.
3. What if someone wants a section from `theta=-45` to `theta=45`? I guess they could go from `theta=45` to `theta=135` and rotate back, but could we allow any `theta` range as long as it spans at most a full circle?

Reply:
1. We should document the allowable ranges whatever this ends up being.

2. My first impulse is always to be wary of the penalty on sorting.  Let the user control garbage in.  But I suspect that it will be small compared to constructing the arrays iteratively, but haven't thought

3. Number 3 only makes sense for `theta` I think.  If we implement this, we'd more likely need to use a `np.isclose` or similar check for a complete circle specification.

Reply:
2. My opinion: I would not mind being forced to provide monotonically increasing input for the arrays and receiving a descriptive error if the input is not monotonically increasing. As long it's documented I think sorting the inputs can be a burden left on the user.

3. I think it would be nice to not force the user to put their data between 0 and 360 for theta. But I agree that phi should continue to be constrained between 0 and 180. On that note though... it might be nice to have a `radians` boolean input so that the user can input the angles in radians if they prefer. But, if the user inputs more than a complete revolution should an error be raised? Or should the input just be truncated to a single revolution with a warning print out?

Reply:
Fair enough. But then let's check that it's sorted (quick `O(N)` vs slow `O(N log N)`) and raise (and document). I suggested sorting because I figured each 1d spanning array will be fairly small.

Reply:
2. is complete.  We now check whether they are monotonically increasing rather than just sorted.  I think this avoids degenerate cells completely.

Reply:
Good call, that's what we should do indeed.

Reply:
All 3 points are now implemented I think.